### ISSUE TITLE
pen.avg_BW not correctly updated  

### ISSUE THREAD

**Issue by tomhuhh** [RuminantFarmSystems/RuFaS#1467]
The attribute `avg_BW` of `pen` is not updated and remains as 0 for all animal categories. 

## Expected Behavior

The attribute `avg_BW` of `pen` is updated on a daily basis.  

## Current Behavior

The attribute `avg_BW` of `pen` is not updated and remains as 0 for all animal categories. 

## Possible Solution

Create a property method in pen that calculates the average body weight of animals in the pen. 

## Steps to Reproduce

1.  First add `avg_BW` to the output by adding a class method in the `animal_module_reporter.py` file. 
2. The class method is as follows. 
```
def report_daily_pen_avg_BW(cls, simulation_day: int, pen_list:List[Pen]) -> None: 
        classname = AnimalModuleReporter.__name__
        funcname = AnimalModuleReporter.report_daily_pen_avg_BW.__name__
        info_map = {
            "class": classname, 
            "function": funcname, 
        }
        for pen in pen_list: 
            variable_to_add = f"{classname}.{funcname}.avg_BW_in_pen_{pen.id}_{pen.animal_combination.name}"
            reference_variable = f"{classname}.{funcname}.avg_BW_in_pen_0_CALF"
            AnimalModuleReporter.data_padder(reference_variable, variable_to_add, 0, simulation_day, info_map)
            om.add_variable(
                 f"avg_BW_in_pen_{pen.id}_{pen.animal_combination.name}",
                pen.avg,
                info_map,
            )
```
3. Use the filter `AnimalModuleReporter.report_daily_pen_avg_BW.avg_BW_in_pen_0_CALF` etc. to output results. 
4. The results returned should be 0 for all animal categories, which means the `avg_BW` attribute is never updated for `pen`.

**Comment by tomhuhh** [comment]
The fact that `pen.avg_BW` is never called or used but the `avg_BW` of animals in a pen is tracked somewhere else in the animal requirement calculations makes Ed and I think that we should check all the attributes of pen and make sure all attributes are correctly updated. The goal is that whenever you call an attribute, it should always be correct and up-to-date. 

This may also be worth considering during the animal module redesign. Copy @JoeWaddell here for future reference.

**Comment by JoeWaddell** [comment]
Since `set_avg_nutrient_rqmts` already exists in the `Pen` class, it may be better to deprecate `avg_BW`, and `avg_DMIest` (and possibly others) that are already represented in  the `AnimalRequirements` dict. 

An other option could be to map the results from `set_requirements` in the `AnimalRequirements` class to their pen attribute counterparts, possibly by simply modifying the `Pen.set_avg_nutrient_rqmts` method to instead modify existing Pen attributes themselves, when appropriate.

**Comment by Isgotkowitz** [comment]
> Since `set_avg_nutrient_rqmts` already exists in the `Pen` class, it may be better to deprecate `avg_BW`, and `avg_DMIest` (and possibly others) that are already represented in the `AnimalRequirements` dict.
> 
> An other option could be to map the results from `set_requirements` in the `AnimalRequirements` class to their pen attribute counterparts, possibly by simply modifying the `Pen.set_avg_nutrient_rqmts` method to instead modify existing Pen attributes themselves, when appropriate.

@tomhuhh correct me if I am wrong, but I remember us looking at the outputs related to `avg_nutrient_rqmts` and for some reason it was not working for calves.

**Comment by JoeWaddell** [comment]
Good point! Calves are handled uniquely in many cases, this being one of them (their ration isn't formulated dynamically at the moment, so the `AnimalRequirements` class isn't used for them). So we'll need to make this solve apply to all classes, or make a separate solution for calves specifically.

After a talk with @tomhuhh and @Isgotkowitz  we decided that I'll add a simple method to calculate average body weight on a daily basis for each pen, instead of solely relying on the metric calculated at each ration interval. `avg_DMIest` may instead be deprecated, since that doesn't need to be recalculated on a daily basis.

**Comment by KFosterReed** [comment]
Hey Team, I love the conversation happening here and wanted to check in with objectives. One challenge we are butting up against currently is the simulation time and adding more daily functions that have to retrieve information from all animals seems like it would be counter to improving the simulation time. If the pen.avg_BW isn't currently used I would like to have a clear justification for needing it before pursuing a solution

**Comment by tomhuhh** [comment]
Hello Kristan! `avg_BW` is currently an attribute of pen but unfortunately it is not properly tracked by the pen and remains 0 regardless of the simulation day. Instead, it is calculated and updated in the animal requirement calculations at each ration interval. We acknowledge the importance of transparent coding practice and efficient usage of the model's computational resources. Thus, the goal here is to standardize the tracking of existing attributes of pen. Our suggestion is that to avoid ambiguity and free up resources in the code, there should be only one `avg_BW`, either we keep it as a pen attribute and correctly track it or remove it from pen. Hope this helps. Thanks!

**Comment by KFosterReed** [comment]
Thanks for the clarification Haowen! I think my question is if there is a proposed use or need for the pen avg_bw? If not, I think deprecating this attribute (as was proposed for the avg_DMI), would also increase clarity/reduce ambiguity. Know that I am not saying we shouldn't move in this direction, just that we should be purposeful.

**Comment by allisterakun** [comment]
Closing this issue as it is fixed in the animal refresh. The "avg_BW" is reported as part of the "average_nutrient_requirements" output and is reported every ration interval.

Use the following filter to confirm:
```json
{
  "name": "avg_bw",
  "filters": [
    "AnimalModuleReporter.report_ration_interval_data.avg_rqmts_pen_[0-9]_LAC_COW"
  ],
  "variables": [
    "avg_BW"
  ]
}
```
